lib/model/model.go

```diff
@@ -189,11 +192,50 @@ func (m *Model) StartFolderRW(folder string) {
 		p.versioner = versioner
 	}
 
+	m.warnAboutOverwritingConfigs(folder)
+
 	m.Add(p)
 
 	l.Okln("Ready to synchronize", folder, "(read-write)")
 }
 
+func (m *Model) warnAboutOverwritingConfigs(folder string) {
+	if m.folderCfgs[folder].ReadOnly {
+		return
+	}
+
+	folderLocation := path.Clean(m.folderCfgs[folder].Path())
+	ignores := m.folderIgnores[folder]
+
+	var filesAtRisk []string
+	for _, protectedFilePath := range m.protectedConfigFiles {
+		// check if file is synced in this folder
+		isSynced := false
+		checkPath := path.Dir(protectedFilePath)
+		for len(checkPath) >= len(folderLocation) {
```

Comment:
You got me excited about eliminating this while loop. Then, the docs reminded me why I didn't use it in the first place: "HasPrefix exists for historical compatibility and should not be used." [0](https://golang.org/pkg/path/filepath/#HasPrefix)